packages/fxa-settings/src/models/reliers/oauth-relier.ts

```diff
@@ -374,5 +405,19 @@ function scopeStrToArray(scopes: string) {
     .trim()
     .split(/\s+|\++/g)
     .filter((x) => x.length > 0);
-  return new Set(arrScopes);
+  return Array.from(new Set(arrScopes));
+}
+
+function replaceItemInArray(
+  array: Array<string>,
+  itemToReplace: string,
+  replaceWith: Array<string>
+) {
+  return [...array.filter((x) => x !== itemToReplace), ...replaceWith];
```

Comment:
I guess the other minor concern with this is&hellip; if the `itemToReplace` string/value isn't in the original `array[]`, the `replaceWith[]` is still appended to the original array.
Maybe not a bug, but possibly an unexpected gotcha.

You could also replace the double destructure w/ `.concat()`:
```js
return array.filter((x) => x !== itemToReplace).concat(...replaceWith);
```

Or if you want a non-miss-concat option, we might have to use `.map()` or `.reduce()`:
```js
function replaceItemInArray((array = [], itemToReplace = "", replaceWith = []) {
  return array.map(item => item === itemToReplace ? replaceWith : item).flat();
}

// HIT
replaceItemInArray(['one', 'two', 'three', 'four'], 'two', ['peter', 'karl']);
// ['one', 'peter', 'karl', 'three', 'four']

// MISS
replacer(['one', 'two', 'three', 'four'], 'seven', ['peter', 'jimmy']);
// ['one', 'two', 'three', 'four']
```

None of the solutions probably handle duplicate values very well, but maybe not worth the minimal effort of piping through `Array.from(new Set(array))` to dedupe.

Reply:
Thanks for scrutinizing this. I like using map with flat and running it through a set. I feel like this definitely better than initial implementation, and much better than my initial pass.